### ISSUE TITLE
use `import type` for type only imports

### ISSUE THREAD

**Issue by RSWilli** [stephenh/ts-proto#367]
TS has the option `"importsNotUsedAsValues": "error"` which results in an error, when an import is only importing types and no actual values (which is the case in svelte TS projects because of esbuild/vitejs):

```
This import is never used as a value and must use 'import type' because 'importsNotUsedAsValues' is set to 'error'.ts(1371)
```

the fix would be to change 

```
import { Oauth } from "foo";
```

to

```
import type { Oauth } from "foo";
```

given that `Oauth` is an interface / type and no actual services are imported

**Comment by stephenh** [comment]
Hey @RSWilli , thanks for filing the issue... You're right, I could see that being a problem.

I started poking around at this, and I think it will be ~tedious but mostly fixable. Any place that ts-proto currently outputs a type name, it will have to specify whether its for just a type, or for a concrete usage...

Can you exclude more of the file that has the import that you want to change? I'm currently wondering if most of the ts-proto output, given it wants to call `${MessageWhatever}.decode/encode` will probably want concrete / non-`type` imports anyway? Are you using like type-only output or something like that?

**Comment by RSWilli** [comment]
@stephenh yes I am using type only output with:

```
...
    --ts_proto_opt=useOptionals=true \
    --ts_proto_opt=outputEncodeMethods=false \
    --ts_proto_opt=outputJsonMethods=false \
    --ts_proto_opt=outputClientImpl=false \
    --ts_proto_opt=outputServices=false \
...
```

```ts
// main file
import { Oauth } from "../../../../ORG/GROUP/PROJECT/proto/grpc_oauth_message"

// ...

export interface Request {
    onlyDemo: boolean
    oauth?: Oauth
}

// ...
```

```ts
// ../../../../ORG/GROUP/PROJECT/proto/grpc_oauth_message

// ...

export interface Oauth {
    token: string
}
```

I also think this problem might only occur in type only output. In that case the solution would be much more simple: if no runtime is exported then use the `t:` prefix as specified in https://github.com/stephenh/ts-poet

**Comment by wgottschalk** [comment]
I'm running into a similar issue. I'll import a type generated from protobuf which relies on importing an enum. I run into the error when I have `"importsNotUsedAsValues": "error"` set in my tsconfig. This is important for compatibility with tools like esbuild/vite/remix etc since the rely on browser modules

**Comment by stephenh** [comment]
@RSWilli / @wgottschalk I'd started working on this as a general "any type import is a type import, any value import is a value import", and it's a big change ...

_However_, if we want to baby-step support this when `onlyTypes` is used, then we could probably add a hack to this line:

https://github.com/stephenh/ts-proto/blob/main/src/utils.ts#L202

Which would be "if `onlyTypes` is set, then _all_ types have the `t:` prefix to use the `import type` output":

```
export function impProto(options: Options, module: string, type: string): Import {
  return imp(`t:${type}@./${module}${options.fileSuffix}`);
}
```

Note the `t:` prefix.

Fwiw this would break a lot of code that wants to do like `FooMessage.encode` or `FooMessage.decode`, but if you're using `onlyTypes` then that code is not output anyway.

If either of you would like to try a PR for this "works only for `onlyTypes`" compromise, it should be pretty easy.

Thanks!

**Comment by stephenh** [comment]
This is fixed in 1.109.0. Thanks @RSWilli !
